system/src/system_control_internal.cpp

```diff
@@ -160,36 +160,32 @@ void SystemControl::processRequest(ctrl_request* req, ControlRequestChannel* /*
         break;
     }
     case CTRL_REQUEST_RESET: {
-        setResult(req, SYSTEM_ERROR_NONE, [](int result, void* data) {
-            HAL_Delay_Milliseconds(1000);
-            System.reset();
-        });
+        HAL_Delay_Milliseconds(1000);
+        auto res = System.reset();
+        setResult(req, res);
         break;
     }
     case CTRL_REQUEST_FACTORY_RESET: {
-        setResult(req, SYSTEM_ERROR_NONE, [](int result, void* data) {
```

Comment:
Actually, no, we can't change these. This will affect the behavior. Note that the lambda is passed as third argument which is a completion handler, so we enter DFU only after the request finishes, otherwise if we return a value here we will immediately execute it and e.g. CLI will fail to get a request response from the device.